contracts/farms/RevoSushiswapFarmBot.sol

```diff
@@ -0,0 +1,78 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "hardhat/console.sol";
+
+import "../sushiswap/interfaces/IMiniChefV2.sol";
+import "./common/RevoUniswapStakingTokenStrategy.sol";
+import "../openzeppelin-solidity/contracts/SafeERC20.sol";
+
+/**
+ * RevoSushiswapFarmBot is a farmbot:
+ *   * that runs on top of an IMiniChefV2 farm
+ *   * whose stakingToken is an IUniswapV2Pair ("LP")
+ *   * that acquires LP constintuent tokens through swaps on an IUniswapV2Router02
+ *   * that mints LP from constituent tokens through an IUniswapV2Router02
+ *
+ * This farmbot is suitable for use on top of Sushiswap yield farms
+ **/
+contract RevoSushiswapFarmBot is RevoUniswapStakingTokenStrategy {
+    using SafeERC20 for IERC20;
+
+    event SushiPidUpdated(address indexed by, uint256 sushiPid);
+
+    IMiniChefV2 public stakingRewards;
+
+    // MiniChefV2 maintains a lists of information on each pool/farm; sushiPid
+    // is used as an index into these lists, and acts as an identifier for which
+    // farm to perform operations against.
+    uint256 public sushiPid;
+
+    constructor(
+        address _owner,
+        address _reserveAddress,
+        address _stakingRewards,
+        address _stakingToken,
+        address _revoFees,
+        uint256 _sushiPid,
+        address[] memory _rewardsTokens,
+        address _router,
+        string memory _symbol
+    )
+        RevoUniswapStakingTokenStrategy(
+            _owner,
+            _reserveAddress,
+            _stakingToken,
+            _revoFees,
+            _rewardsTokens,
+            _router,
+            _router,
+            _symbol
+        )
+    {
+        stakingRewards = IMiniChefV2(_stakingRewards);
+        sushiPid = _sushiPid;
+    }
+
+    function updateSushiPid(uint256 _sushiPid)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        sushiPid = _sushiPid;
```

Comment:
if the person updating this were dishonest, could they divert funds to a fake sushi rewards contract that they can withdraw from at will?

Reply:
relatedly, is there any scenario you can imagine us wanting to update this field?

I guess if a new rewards contract for the same pair was created we could switch it over when the old contract is depleted. Is that what you were thinking?

Reply:
Yes, that was the intention. In fact, this makes me wonder if we should implement some process by which farmbots can be completely transitioned to using a different underlying staking rewards contract (that use the same staking token) without forcing users to swap out of the farm/withdraw their LP. Some sort of process by which we:

1. Pause the farm, withdrawing all LP so that it's still managed by the farm bot. 
2. Update the necessary parameters in order to change the underlying farm (staking rewards contract address, rewards tokens, etc.) These parameters should presumably only be changeable when the contract is paused.
3. Unpause the contract, automatically staking all held LP into the new farm.

Might be worth looking into, and could preemptively solve issues down the line.